internal/dag/builder.go

```diff
@@ -919,6 +923,27 @@ func (b *Builder) processTCPProxy(sw *ObjectStatusWriter, ir *ingressroutev1.Ing
 	}
 }
 
+func (b *Builder) processTCPProxyHTTPProxy(sw *ObjectStatusWriter, httpproxy *projcontour.HTTPProxy, host string) {
+	if len(httpproxy.Spec.TCPProxy.Services) > 0 {
+		var proxy TCPProxy
+		for _, service := range httpproxy.Spec.TCPProxy.Services {
+			m := Meta{name: service.Name, namespace: httpproxy.Namespace}
+			s := b.lookupService(m, intstr.FromInt(service.Port))
+			if s == nil {
+				sw.SetInvalid(fmt.Sprintf("tcpproxy: service %s/%s/%d: not found", httpproxy.Namespace, service.Name, service.Port))
+				return
+			}
+			proxy.Clusters = append(proxy.Clusters, &Cluster{
+				Upstream:             s,
+				LoadBalancerStrategy: service.Strategy,
+			})
+		}
+		b.lookupSecureVirtualHost(host).TCPProxy = &proxy
+		sw.SetValid()
```

Comment:
please drop this, set valid is called on like 628.

Reply:
👍